pennylane_lightning/core/src/simulators/lightning_qubit/bindings/LQubitBindings.hpp

```diff
@@ -259,6 +259,19 @@ void registerBackendSpecificMeasurements(PyClass &pyclass) {
                  const std::string &, const std::vector<std::size_t> &)>(
                  &Measurements<StateVectorT>::expval),
              "Expected value of an operation by name.")
+        .def(
+            "expval",
+            [](Measurements<StateVectorT> &M, const np_arr_c &matrix,
+               const std::vector<std::size_t> &wires) {
+                const std::size_t matrix_size =
+                    Pennylane::Util::exp2(2 * wires.size());
```

Comment:
```suggestion
                const std::size_t matrix_size = matrix.size;
```
Also, I suggest to add a assert with PL_ABORT if the matrix size don't match with `Pennylane::Util::exp2(2 * wires.size());`